src/twisted/python/compat.py

```diff
@@ -150,6 +151,20 @@
 )
 
 
+def md5(data=b"", **kwargs):
+    """
+    Wrapper around hashlib.md5
```

Comment:
I don't understand this.

My understanding is that when FIPS mode is enabled, MD5 usage should fail.

----------

For Twisted, I think the goal is replace the usage of md5 with SHA256 in as many places as possible.

Where MD5 usage is  required, since this is how the network protocol works, then those test should expect to fail, and they  should be skipped when running on FIPS

Reply:
ok.... so the scope of this PR is to have Twisted  work with a FIPS OpenSSL for which the FIPS mode is disabled... is that correct ? :)

As mentioned by Glyph.
Just setting up the automated  testing environment is a huge improvement over what we have in `trunk` and is enough to  get the code merged.

Reply:
> the scope of this PR is to have Twisted work with a FIPS OpenSSL for which the FIPS mode is disabled... is that correct ?

No with this patch, twisted will work the same way it works in non fips environment. Users will not see any difference in the way twisted works even when fips is enabled.

When we install openssl-fips-provider package in a Photon's docker container image, openssl fips gets enabled by default.

Try the below steps if you want to see it in action:
```
docker run -it --rm photon:5.0 /bin/bash

Inside the container:

tdnf install -y openssl-fips-provider

# the below command should fail
openssl md5 /etc/os-release


# Also, below command will show 1 as output. (but need python >= 3.9)
python3 -c "import _hashlib; print('fips mode: ', _hashlib.get_fips_mode())"
```

`usedforsecurity` flag in hashlib module was introduced to temporarily allow fips incompatible algorithms so that things will keep working when there is a hard requirement for md5 like algorithms. This PR does exactly that.

With this fix, Twisted will work the same way when fips is enabled or disabled.

Reply:
> > the scope of this PR is to have Twisted work with a FIPS OpenSSL for which the FIPS mode is disabled... is that correct ?
> 
> No with this patch, twisted will work the same way it works in non fips environment. Users will not see any difference in the way twisted works even when fips is enabled.
> 

It's a really great practice to make this kind of information very clear on the issue.  That way reviews don't have to guess the intent of the changes or discover it buried in a comment deep in the middle of the PR.